src/main/java/seedu/address/model/person/Appointment.java

```diff
@@ -0,0 +1,116 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_DATE_FORMAT;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a person's Appointment in the address group.
+ */
+public class Appointment {
+    public static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm");
+    public static final String NO_APPOINTMENT_SCHEDULED = "No appointment scheduled!";
+    public static final String MESSAGE_CONSTRAINTS = "Appointment dates have to be of format dd-MM-yyyy HHmm!";
+
+    private final LocalDateTime appointmentDate;
+
+    /**
+     * Constructs an appointment with a given date.
+     *
+     * @param appointmentDate Date of the appointment.
+     */
+    public Appointment(String appointmentDate) {
+        requireNonNull(appointmentDate);
+        checkArgument(isValidDate(appointmentDate), MESSAGE_INVALID_DATE_FORMAT);
+        this.appointmentDate = LocalDateTime.parse(appointmentDate, DATE_FORMAT);
+    }
+
+    /**
+     * Constructs an empty Appointment.
+     */
+    public Appointment() {
+        this.appointmentDate = null;
+    }
+
+    /**
+     * Returns true if date is valid.
+     *
+     * @param testDate Date to be tested.
+     * @return True if valid.
+     */
+    public static boolean isValidDate(String testDate) {
+        try {
+            LocalDateTime.parse(testDate, DATE_FORMAT);
+        } catch (DateTimeParseException e) {
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Returns true is appointment is valid.
+     *
+     * @param test Input args for appointment test.
+     * @return True if valid.
+     */
+    public static boolean isValidAppointment(String test) {
+        return test.equals(NO_APPOINTMENT_SCHEDULED) || isValidDate(test);
+    }
+
+    /**
+     * Factory method for the construction of an appointment from the JSON storage file.
+     *
+     * @param in Json storage format of an appointment.
+     * @return An appointment with the input field.
+     */
+    public static Appointment of(String in) {
+        if (in.equals(NO_APPOINTMENT_SCHEDULED)) {
+            return new Appointment();
+        } else {
+            return new Appointment(in);
+        }
+    }
```

Comment:
1. Is this factory method exclusively used for JSON storage file reading? Also, if you are using a factory method, should the constructors of `Appointment` be exposed / private? 

2. For invalid input passed to this method, I understand that an exception is thrown by the constructor. However, will this exception cause the program to default to an empty address because of the invalid appointment? Is it possible to handle this case by giving an empty appointment when for invalid input (while returning an error string to the user) such that the whole program because of a single appointment.

Reply:
1. The factory method is exclusively used for the storage file reading since from the data file, it can only have 2 inputs, either a valid appointment date or the NO_APPOINTMENT_SCHEDULED string. I still use the Appointment constructors in the some of the other files, like in the actual command for adding appointments, but I can swap to using the factory method exclusively as well.

2. The control flow of the program with invalid inputs, similar to the other commands, will simply break the control flow without having any input into the system, such as for the add record/ add patient commands. I feel there shouldn't be a positive response (i.e adding an empty appointment) if the input into the system is wrong, since we should not be validating any invalid input.

Reply:
I agree with marclzh for point 2. Invalid inputs to commands should not do anything and should only display an error message to the user and only valid commands should have an effect on the program.

Reply:
Regarding @EiffelLKF's first point, I have updated the privacy modifiers of the Appointment class. New Appointment objects now must utilise the factory method for generation.